docs/topics/rotate-certs.md

```diff
@@ -0,0 +1,127 @@
+
+## Under The Hood
+
+A Kubernetes cluster relies on multiple PKIs to secure the communication between its components (apiserver, kubelet, etcd, etc). On an AKS Engine cluster, these multiple PKIs share a single certificate authoritiy (CA). In order to reduce downtime, `aks-engine rotate-certs` rotates each of these PKIs (etcd, service account tokens, apiserver-kubelet) in sequence.
+
+Each of these are idempotent discrete steps. The cluster is expected to reach a healthy state after each step is executed. If the certificate rotation process halts before completion due to a transient issue (e.x.: network connectivity), rerun `aks-engine rotate-certs` using the `--continue` flag so the process can be resumed at the point where it failed.
+
+At a high level, the `aks-engine rotate-certs` command performs the following tasks:
+
+- backup current set of certificates in directory `_rotate_certs_backup/` (relative to the `apimodel.json` path)
+- generate/load new set of certificate and persist them in local directory `_rotate_certs_output/` (relative to the `apimodel.json` path)
+- distribute certificates
+- rotate certificates
+- update input `apimodel.json` with new certificates information
+
+### Generating certificates
+
+If flag `--generate-new-certificates` is set, AKS Engine generates the new set of certificates that will be deployed to the cluster.
+
+Alternatively, AKS Engine can load a new set of certificates from a JSON file specified in `--certificate-profile`.
+
+```json
+{
+  "certificateProfile": {
+    "caCertificate": "",
+    "caPrivateKey": "",
+    "apiServerCertificate": "",
+    "apiServerPrivateKey": "",
+    "clientCertificate": "",
+    "clientPrivateKey": "",
+    "kubeConfigCertificate": "",
+    "kubeConfigPrivateKey": "",
+    "etcdServerCertificate": "",
+    "etcdServerPrivateKey": "",
+    "etcdClientCertificate": "",
+    "etcdClientPrivateKey": "",
+    "etcdPeerCertificates": [],
+    "etcdPeerPrivateKeys": []
+  }
+}
+```
+
+### Certificates distribution
+
+The new certificates are securely copied to each cluster node before the certificates rotation process starts. On Linux nodes, they are located in directory `/tmp/akse/certs`. On Windows nodes, the directory is `$env:temp`.
+
+## Best Practices
+
+### Use a reliable network connection
+
+`aks-engine rotate-certs` requires the execution of multiple remote commands which are subject to potential failures, mostly if the connection to the cluster nodes is not reliable.
+
+Executing `aks-engine rotate-certs` from a VM running on the target cloud (Azure or Azure Stack) can drastically reduce the occurence of transient issues.
+
+## Known Limitations
+
+### Cluster-autoscaler
+
+`aks-engine rotate-certs` will be blocked on deployments that include the `cluster-autoscaler` addon. Because the cluster certificates are embedded in the ARM template, the addon won't be able to produce functioning agent nodes after a certificate rotation operation.
```

Comment:
I think it's safe to assume that AKS Engine + cluster-autoscaler only works on VMSS node pools, in which case it's more correct to say that the "cluster certificates are embedded in the VMSS model".

Also, we should probably just disable cluster-autoscaler during this operation, and re-enable afterwards, right?

Reply:
taking that idea